designs/modify-resources-attached-to-code-pipeline.md

```diff
@@ -0,0 +1,163 @@
+
+Proposed:
+After the wizard has collected all information from the user, we can check to see if the Stack belongs to a CodePipeline. If it does not, the deploy proceeds. If it does, the user is given a confirmation prompt (see [Confirmation Prompt](#confirmation-prompt) below). If the user agrees to proceed with the deploy, it proceeds normally. Otherwise, nothing happens - the wizard has already ended, and the user is back in the editor.
+
+Alternate:
+After the user enters a CloudFormation Stack, but before the wizard closes, we can check to see if the Stack belongs to a CodePipeline. If it does not, the wizard continues. If it does, the user is given a confirmation prompt, however this confirmation prompt is integrated with the wizard. If the user agrees to proceed with the deploy, the wizard moves on to the next step. Otherwise, the wizard moves back to the step where the user can specify a different CloudFormation Stack.
+
+### <a id="confirmation-prompt"></a>Confirmation Prompt
+
+The confirmation prompt contains the following messaging. See [Success criteria](#success-criteria) for the required information.
+
+---
+
+{ AWS Lambda Function | AWS CloudFormation Stack } '`arn_here`' is part of an AWS CodePipeline pipeline (`pipeline_arn_here`).
+
+Manual modification of resources outside of an AWS CodePipeline will skip any verification steps defined in the pipeline.
+
+Do you want to continue with this manual { deployment | deletion }?
+
+---
+
+The user will have the following actions/responses available:
+* `Yes` - proceeds with the manual operation
+* `No` - The default - Cancels the current operation (or goes back a step in the case of a wizard)
+* `Learn More` - This will link to user documentation (https://docs.aws.amazon.com/codepipeline/latest/userguide) but will not close the confirmation prompt
+
+Implementation
+==============
+
+Design
+------
+
+### Resources associated with a pipeline
+
+Resources that have a tag `aws:codepipeline:pipelineArn` are considered to have an association with a pipeline.
+
+### Utility Methods
+* a method that determines if an AWS resource is associated with AWS CodePipeline
+* a method that generates confirmation prompt text (if confirmation prompt will exist within a Wizard and as a standalone prompt)
+* a method that encapsulates showing users the (non-wizard) confirmation prompt
+
+### Confirmation Prompt
+
+The confirmation prompt is a QuickPick instead of a message box.
+* The message box can be "dismissed" with the `Esc` key, which hides it in a "message box queue" rather than cancelling the message. The user flow appears to be cancelled, but in reality it is suspended (until the queued message box is restored, and a button is pressed).
```

Comment:
This is confusing to me...I'm assuming this means that this will function as a "back" button when using a QuickPick wizard workflow? Will this function the same as clicking "No"? If not, should it?

Reply:
Rewriting for clarity